pkg/operator/collection.go

```diff
@@ -503,6 +503,42 @@ func (r *collectionReconciler) makeCollectorConfig(ctx context.Context, spec *mo
 			continue
 		}
 		cfg.ScrapeConfigs = append(cfg.ScrapeConfigs, cfgs...)
+
+		change, err := nm.Status.SetMonitoringCondition(nm.GetGeneration(), metav1.Now(), cond)
+		if err != nil {
+			// Log an error but let operator continue to avoid getting stuck
+			// on a potential bad resource.
+			logger.Error(err, "setting nodemonitoring status state")
+		}
+
+		if change {
+			// convert ScrapeNodeEndpoint to ScrapeEndpoint.
+			var scrapeEndpoints []monitoringv1.ScrapeEndpoint
+			for _, endpoint := range nm.Spec.Endpoints {
+				scrapeEndpoints = append(scrapeEndpoints,
+					monitoringv1.ScrapeEndpoint{
+						Scheme:           endpoint.Scheme,
+						Path:             endpoint.Path,
+						Params:           endpoint.Params,
+						Interval:         endpoint.Interval,
+						Timeout:          endpoint.Timeout,
+						MetricRelabeling: endpoint.MetricRelabeling},
+				)
+			}
+			nodeMonConv := monitoringv1.PodMonitoring{
+				ObjectMeta: nm.ObjectMeta,
+				TypeMeta:   nm.TypeMeta,
+				Spec: monitoringv1.PodMonitoringSpec{
+					Selector:  nm.Spec.Selector,
+					Endpoints: scrapeEndpoints,
+					Limits:    nm.Spec.Limits,
+				},
+				Status: monitoringv1.PodMonitoringStatus{
+					MonitoringStatus: nm.Status,
+				},
+			}
+			r.statusUpdates = append(r.statusUpdates, &nodeMonConv)
```

Comment:
issue: I'm not sure if this is right. `statusUpdates` is for updating statuses of _existing_ [Cluster]PodMonitorings but one doesn't exist here. I believe your intention was to pass the NodeMonitoring object here.  An e2e test would have caught this.

I would rename `PodMonitoringCRD.GetStatus` to something like `PodMonitoringCRD.GetPodMonitoringStatus` and create a new interface with a method `MonitoringCRD.GetStatus` that returns a `MonitoringStatus`. Then, `statusUpdates` can store a `MonitoringCRD` and you'd be able to pass in your `NodeMonitoring` here.